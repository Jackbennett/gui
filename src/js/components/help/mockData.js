export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
