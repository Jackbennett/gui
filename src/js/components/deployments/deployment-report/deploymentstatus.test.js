import React from 'react';
import { render } from '@testing-library/react';
import RolloutSchedule from './rolloutschedule';
import { defaultState, undefineds } from '../../../../../tests/mockData';

describe('RolloutSchedule Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<RolloutSchedule deployment={defaultState.deployments.byId.d2} innerRef={jest.fn} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
