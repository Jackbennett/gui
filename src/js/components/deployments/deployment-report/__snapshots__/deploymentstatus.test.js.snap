// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolloutSchedule Component renders correctly 1`] = `
<div>
  <h4
    class="dashboard-header margin-top-large"
  >
    <span>
      Schedule details
    </span>
  </h4>
  <div
    class="flexbox"
  >
    <div
      class="break-all two-columns column-data  "
      style="align-items: center; align-self: flex-start; grid-template-columns: minmax(140px, 1fr) minmax(220px, 1fr); max-width: 25vw;"
    >
      <div
        class="align-right  text-muted"
      >
        <b>
          Start time
        </b>
      </div>
      <div
        class="flexbox "
      >
        <time
          datetime="2019-01-01T12:25:00+00:00"
        >
          2019-01-01 12:25
        </time>
      </div>
      <div
        class="align-right  text-muted"
      >
        <b>
          Current phase
        </b>
      </div>
      <div
        class="flexbox "
      >
        1
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="margin-left: 32px; margin-right: 32px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
    <div
      class="break-all two-columns column-data  "
      style="align-items: center; align-self: flex-start; grid-template-columns: minmax(140px, 1fr) minmax(220px, 1fr); max-width: 25vw;"
    >
      <div
        class="align-right  text-muted"
      >
        <b>
          End time
        </b>
      </div>
      <div
        class="flexbox "
      >
        -
      </div>
    </div>
  </div>
  <div
    class="progress-chart-container margin-top"
  >
    <div
      class="progress-chart detailed"
    >
      <div
        class="progress-step progress-step-total"
      >
        <div
          class="progress-bar"
        />
      </div>
      <div
        class="progress-step"
        style="flex-grow: 1;"
      >
        <div
          class="progress-step-number text-muted"
        >
          Phase 1
        </div>
        <div
          class="flexbox progress-bar"
        >
          <div
            class="progress-bar green"
            style="width: 0%;"
          />
          <div
            class="progress-bar warning"
            style="width: 0%;"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="deployment-phases-report margin-top margin-bottom"
    style="grid-template-columns: repeat(auto-fit, 270px);"
  >
    <div
      class="flexbox column centered"
      style="max-width: 270px; border-radius: 5px; padding-top: 8px; padding-bottom: 24px;"
    >
      <div
        class="text-muted"
      >
        Complete
      </div>
      <div
        class="MuiChip-root MuiChip-sizeSmall"
        style="align-self: flex-start; margin: 16px 16px 16px 20px;"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          Phase 1
        </span>
      </div>
      <div
        class="break-all two-columns column-data  "
        style="align-items: center; align-self: initial; grid-template-columns: minmax(140px, 1fr) minmax(220px, 1fr); max-width: 25vw;"
      >
        <div
          class="align-right  text-muted"
        >
          <b>
            Start time
          </b>
        </div>
        <div
          class="flexbox "
        >
          <time
            datetime="2019-01-01T12:25:00+00:00"
          >
            2019-01-01 12:25
          </time>
        </div>
        <div
          class="align-right  text-muted"
        >
          <b>
            Batch size
          </b>
        </div>
        <div
          class="flexbox "
        >
          <div
            class="text-muted"
          >
            100% (1 device)
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
